src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -353,4 +354,23 @@ public static void deleteCollection(Firestore database, String collectionToDelet
       throw new FirebaseException(e.toString());
     }
   }
+
+  /**
+   * Queries documents if one of their array fields contain a certain value.
+   * @param database
+   * @param collectionName
+   * @param arrayName
+   * @param arrayFieldValue
+   * @return All documents in database collection specified by collectionName that containes
+   * arrayFieldValue in their arrayName field, which is supposed to be an array.
+   * @throws FirebaseException
+   * @throws NoSuchElementException
+   */
+  public static String queryByArrayField(Firestore database, String collectionName, String arrayName,  
+      Optional<String> arrayFieldValue) throws FirebaseException, NoSuchElementException {
```

Comment:
So if I throw all possible errors in this way, I don't need to catch them, right?

Reply:
Right, but somewhere up the call chain, you'll need to handle them.

Reply:
Yeah I handled them in endpoint servlets/